public/app/features/alerting/unified/components/receivers/form/fields/OptionField.tsx

```diff
@@ -125,14 +125,6 @@ const OptionInput: FC<Props & { id: string }> = ({
   const secureFieldKey = option.secure && option.secureFieldKey ? option.secureFieldKey : '';
   const isEncryptedInput = secureFieldKey && secureFields?.[secureFieldKey];
 
-  // workaround for https://github.com/react-hook-form/react-hook-form/issues/4993#issuecomment-829012506
```

Comment:
👍

Reply:
This workaround was causing unexpected race conditions when different notifiers used the same setting names (for example, the `url` field). The unregister call here happens inside `useEffect`, which runs fairly late in the render cycle (after the render, actually), and it was overwriting values of the newly selected contact point